src/librustc/util/common.rs

```diff
@@ -29,6 +33,59 @@ pub struct ErrorReported;
 
 thread_local!(static TIME_DEPTH: Cell<usize> = Cell::new(0));
 
+/// Initialized for -Z profile-queries
+thread_local!(static PROFQ_CHAN: RefCell<Option<Sender<ProfileQueriesMsg>>> = RefCell::new(None));
```

Comment:
Good question.  The separate thread design was @nikomatsakis's suggestion, so I'll let him respond first.